packages/flutter/lib/src/widgets/route_notification_messages.dart

```diff
@@ -23,6 +33,28 @@ class RouteNotificationMessages {
     }
   }
 
+  /// Notifies the platform of a route information change.
+  ///
+  /// See also:
+  ///
+  ///  * [SystemChannels.router], which handles subsequent router requests
+  static void _notifyRouteInformationChange(RouteInformation routeInformation, RouteInformation previousRouteInformation) {
+    SystemChannels.router.invokeMethod<void>(
+      'routeInformationUpdated',
+      <String, dynamic>{
+        'previousRouteName': _convertRouteInformationToMap(previousRouteInformation),
+        'routeName': _convertRouteInformationToMap(routeInformation),
```

Comment:
This is more than just a routeName.

Also, why do we need to send the old information as well? If the engine needs it, it should keep a copy of it.

Reply:
Good question, I remember the last time i check the previous one is not used and discarded. I probably should just remove it.